Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -386,9 +394,184 @@ namespace Tensile
                 else
                     m_fanValues[i] += newValue;
             }
+            
+            rsmi_gpu_metrics_t gpuMetrics;
+            auto status = rsmi_dev_gpu_metrics_info_get(m_smiDeviceIndex, &gpuMetrics);
+            if(status != RSMI_STATUS_SUCCESS)
+            {
+                m_freqValues.at(0)        = std::numeric_limits<uint16_t>::max();
+                m_powerValues.at(0)       = std::numeric_limits<uint16_t>::max();
+                m_tempHotspotValues.at(0) = std::numeric_limits<uint16_t>::max();
+            }
+            else
+            {
+                // size of m_freqValues,m_powerValues,m_tempHotspotValues is same.   
+                if(m_freqValues[0] != std::numeric_limits<uint16_t>::max())  
+                {    
+                    m_freqValues.push_back(gpuMetrics.average_gfxclk_frequency);
```

Comment:
This is not a good solution.
We should remove the first element if we get the valid result.

I think it is better to have another variable to specify if the result is valid or not.
(instead of using [0] to identify if it is valid or not)

Reply:
Or, just consider size()==0 is invalid.

Reply:
I am not sure if this is called multiple times.
If we call this multiple times and make the result invalid if one of the calls is invalid (like valid, valid, invalid -> invalid), it is better to have another variable.
If not, I think we just need to push the result only if the result is valid (do nothing here if the result is invalid).

Reply:
added another variable to keep track invalid GPU index